src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/YFlowCrudSpec.groovy

```diff
@@ -36,122 +51,498 @@ class YFlowCrudSpec extends HealthCheckSpecification {
-        given: "A y-flow"
-        assumeTrue(topology.activeTraffGens.size() >= 2,
-"There should be at least two active traffgens for test execution")
-        def traffExam = traffExamProvider.get()
-        def allLinksInfoBefore = northbound.getAllLinks().collectEntries { [it.id, it.availableBandwidth] }.sort()
-        flowHelperV2.addFlow(flow)
-        def path = PathHelper.convert(northbound.getFlowPath(flow.flowId))
-        def switches = pathHelper.getInvolvedSwitches(path)
-        //for single-flow cases need to add switch manually here, since PathHelper.convert will return an empty path
-        if (flow.source.switchId == flow.destination.switchId) {
-            switches << topology.activeSwitches.find { it.dpId == flow.source.switchId }
+    def "Valid y-flow can be created#trafficDisclaimer, covered cases: #coveredCases"() {
+        assumeTrue(swT != null, "These cases cannot be covered on given topology: $coveredCases")
+
+        when: "Create a y-flow of certain configuration"
+        def yFlow = northboundV2.addYFlow(yFlowRequest)
+
+        then: "Y-flow is created and has UP status"
+        Wrappers.wait(WAIT_OFFSET) { assert northboundV2.getYFlow(yFlow.YFlowId).status == FlowState.UP.toString() }
+
+        and: "2 sub-flows are created, visible via regular 'dump flows' API"
+        northboundV2.getAllFlows()*.flowId.sort() == yFlow.subFlows*.flowId.sort()
+
+        and: "History has relevant entries about y-flow creation"
+        northbound.getFlowHistory(yFlow.YFlowId).last().payload.last().action == CREATE_SUCCESS_Y
+        yFlow.subFlows.each {
+            assert northbound.getFlowHistory(it.flowId).last().payload.last().action == CREATE_SUCCESS
+        }
+
+        and: "User is able to view y-flow paths"
+        def paths = northboundV2.getYFlowPaths(yFlow.YFlowId)
+
+//        and: "Y-flow passes flow validation"
+//        northbound.validateFlow(yFlow.YFlowId)
+
+        and: "Both sub-flows pass flow validation"
+        yFlow.subFlows.each {
+            assert northbound.validateFlow(it.flowId).every { it.asExpected }
         }
 
-        expect: "No rule discrepancies on every switch of the flow"
-        wait(WAIT_OFFSET) { //due to instability on jenkins in multiTable mode + server42
-            switches.each { verifySwitchRules(it.dpId) }
+        and: "All involved switches pass switch validation"
+        def involvedSwitches = pathHelper.getInvolvedSwitches(paths)
+        involvedSwitches.each { sw ->
+            northbound.validateSwitch(sw.dpId).verifyRuleSectionsAreEmpty(sw.dpId, ["missing", "excess", "misconfigured"])
+            northbound.validateSwitch(sw.dpId).verifyMeterSectionsAreEmpty(sw.dpId, ["missing", "excess", "misconfigured"])
         }
 
-        and: "No discrepancies when doing flow validation"
-        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+//        and: "Bandwidth is properly consumed on shared and non-shared ISLs"
```

Comment:
can we ping the whole y-flow? or just sub-flow?

Reply:
not sure as of now. will check